packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -660,8 +661,21 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
         'redshift', 'redshift-data', 's3', 'sagemaker.api', 'sagemaker.featurestore-runtime', 'sagemaker.runtime', 'securityhub',
         'servicecatalog', 'sms', 'sqs', 'states', 'sts', 'sync-states', 'synthetics', 'transcribe', 'transcribestreaming', 'transfer',
         'workspaces', 'xray'],
+      'us-isof-': ['ecr.api', 'ecr.dkr'],
+      'eu-isoe-': ['ecr.api', 'ecr.dkr'],
     };
-    if (VPC_ENDPOINT_SERVICE_EXCEPTIONS[region]?.includes(name)) {
+
+    const regionPartition = region.split('-').slice(0, 2).join('-');
+    const partitionDetails = PARTITION_MAP[`${regionPartition}-`];
```

Comment:
Does this work for commercial regions? The map seems to use `default` for cmmercial regions, but your `regionPartition` won't return `default` for commercial regions.

Reply:
this is not for commercial regions, only for isolated regions that too only for the ones raised in the request with expected endpoint, as don't want to impact any other region . For regions/services other than the one defined in exception list it will be `com.amazonaws` i.e commercial region too

Reply:
I see, that makes sense. THanks